pkg/tests/upgrade_tests/upgrade_test.go

```diff
@@ -60,16 +62,22 @@ func TestMain(m *testing.M) {
 
 /* Prev image is the db image with the old promscale extension. We do NOT test timescaleDB extension upgrades here. */
 func getDBImages(extensionState testhelpers.ExtensionState) (prev string, clean string) {
-	switch extensionState {
-	case testhelpers.MultinodeAndPromscale:
-		return "timescaledev/promscale-extension:0.1.1-ts2-pg12", "timescaledev/promscale-extension:latest-ts2-pg12"
-	case testhelpers.Timescale1AndPromscale:
-		return "timescaledev/promscale-extension:0.1.1-ts1-pg12", "timescaledev/promscale-extension:latest-ts1-pg12"
-	case testhelpers.Timescale1:
-		return "timescale/timescaledb:latest-pg12", "timescale/timescaledb:latest-pg12"
-	default:
-		panic("Unexpected extension state in upgradeTest")
+	if extensionState.UsesPG12() {
+		if extensionState.UsesMultinode() {
+			return "timescaledev/promscale-extension:0.1.1-ts2-pg12", "timescaledev/promscale-extension:latest-ts2-pg12"
+		}
+		if extensionState.UsesTimescaleDB() {
+			if extensionState.UsesTimescale2() {
+				return "timescaledev/promscale-extension:0.1.1-ts2-pg12", "timescaledev/promscale-extension:latest-ts2-pg12"
+			} else {
+				return "timescaledev/promscale-extension:0.1.1-ts1-pg12", "timescaledev/promscale-extension:latest-ts1-pg12"
+			}
+		}
+		if !extensionState.UsesTimescaleDB() {
+			return "timescale/timescaledb:latest-pg12", "timescale/timescaledb:latest-pg12"
+		}
 	}
+	panic("Unexpected extension state in upgradeTest")
```

Comment:
Seems like we could panic first here, would make this condition tree a bit more easier to read.

Else is not needed since we use return statements and I would prefer a switch statement honestly. 
```suggestion
	if !extensionState.UsesPG12() {
		panic("Unexpected extension state in upgradeTest")
	}
	switch {
	case extensionState.UsesMultinode():
		return "timescaledev/promscale-extension:0.1.1-ts2-pg12", "timescaledev/promscale-extension:latest-ts2-pg12"
	case !extensionState.UsesTimescaleDB():
		return "timescale/timescaledb:latest-pg12", "timescale/timescaledb:latest-pg12"
 	case extensionState.UsesTimescale2() :
		return "timescaledev/promscale-extension:0.1.1-ts2-pg12", "timescaledev/promscale-extension:latest-ts2-pg12"
	default:
		return "timescaledev/promscale-extension:0.1.1-ts1-pg12", "timescaledev/promscale-extension:latest-ts1-pg12"
	}
```